tests/test_framework/bitcoind.py

```diff
@@ -0,0 +1,200 @@
+        return f
+
+
+class Bitcoind(TailableProc):
+    def __init__(self, bitcoin_dir, rpcport=None):
+        TailableProc.__init__(self, bitcoin_dir, verbose=False)
+
+        if rpcport is None:
+            rpcport = reserve()
+
+        self.bitcoin_dir = bitcoin_dir
+        self.rpcport = rpcport
+        self.p2pport = reserve()
+        self.prefix = "bitcoind"
+
+        regtestdir = os.path.join(bitcoin_dir, "regtest")
+        if not os.path.exists(regtestdir):
+            os.makedirs(regtestdir)
+
+        self.cmd_line = [
+            "bitcoind",
+            "-datadir={}".format(bitcoin_dir),
+            "-printtoconsole",
+            "-server",
+            "-logtimestamps",
+            "-rpcthreads=16",
+        ]
+        bitcoind_conf = {
+            "port": self.p2pport,
+            "rpcport": rpcport,
+            "debug": 1,
+            "fallbackfee": Decimal(1000) / COIN,
+        }
+        self.conf_file = os.path.join(bitcoin_dir, "bitcoin.conf")
+        with open(self.conf_file, "w") as f:
+            f.write("chain=regtest\n")
+            f.write("[regtest]\n")
+            for k, v in bitcoind_conf.items():
+                f.write(f"{k}={v}\n")
+
+        self.rpc = BitcoinDProxy(bitcoin_dir, "regtest", rpcport)
+
+    def start(self):
+        TailableProc.start(self)
+        self.wait_for_log("Done loading", timeout=TIMEOUT)
+
+        logging.info("BitcoinD started")
+
+    def stop(self):
+        self.rpc.stop()
+        return TailableProc.stop(self)
+
+    # wait_for_mempool can be used to wait for the mempool before generating
+    # blocks:
+    # True := wait for at least 1 transation
+    # int > 0 := wait for at least N transactions
+    # 'tx_id' := wait for one transaction id given as a string
+    # ['tx_id1', 'tx_id2'] := wait until all of the specified transaction IDs
+    def generate_block(self, numblocks=1, wait_for_mempool=0):
+        if wait_for_mempool:
+            if isinstance(wait_for_mempool, str):
+                wait_for_mempool = [wait_for_mempool]
+            if isinstance(wait_for_mempool, list):
+                wait_for(
+                    lambda: all(
+                        txid in self.rpc.getrawmempool() for txid in wait_for_mempool
+                    )
+                )
+            else:
+                wait_for(lambda: len(self.rpc.getrawmempool()) >= wait_for_mempool)
+
+        old_blockcount = self.rpc.getblockcount()
+        addr = self.rpc.getnewaddress()
+        self.rpc.generatetoaddress(numblocks, addr)
+        wait_for(lambda: self.rpc.getblockcount() == old_blockcount + numblocks)
+
+    def create_utxo(self, address, amount):
+        """Send {amount} coins to this {address}, return the outpoint of the
+        created UTXO."""
+        txid = self.rpc.sendtoaddress(address, amount)
+        vout = 0
+        while True:
+            txoinfo = self.rpc.gettxout(txid, vout)
+            if txoinfo["scriptPubKey"]["address"] == address:
```

Comment:
`"address"` gave a `KeyError`. I changed to 
```python
if txoinfo["scriptPubKey"]["addresses"][0] == address:
```
to fix it (as per the [docs](https://bitcoincore.org/en/doc/0.21.0/rpc/blockchain/gettxout/))